src/core/utils/FocusTrap.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { getFocusableElements } from './common';
+import { mergeRefs } from './hooks/useMergedRefs';
+
+export type FocusTrapProps = {
+  /**
+   * A single child element to trap focus in.
+   */
+  children: JSX.Element;
+};
+
+/**
+ * Trap focus within child container.
+ * Used for modals and dialogs.
+ */
+export const FocusTrap = (props: FocusTrapProps) => {
+  const { children } = props;
+  const childRef = React.useRef<HTMLElement>();
+
+  const onLastFocus = (event: React.FocusEvent) => {
+    const elements = getFocusableElements(childRef.current);
+    const firstElement = elements[0];
+    const lastElement = elements[(elements.length || 1) - 1];
+
+    if (event.relatedTarget === lastElement) {
+      (firstElement as HTMLElement).focus();
+    } else {
+      (lastElement as HTMLElement).focus();
+    }
+  };
+
+  const onFirstFocus = (event: React.FocusEvent) => {
+    const elements = getFocusableElements(childRef.current);
+    const firstElement = elements[0];
+    const lastElement = elements[(elements.length || 1) - 1];
+
+    if (
+      event.relatedTarget === childRef.current ||
+      event.relatedTarget === firstElement
+    ) {
+      (lastElement as HTMLElement).focus();
+    } else {
+      (firstElement as HTMLElement).focus();
+    }
+  };
+
+  return (
+    <div>
```

Comment:
Do you need this extra div or can you just use fragment?

Reply:
I addded it, cause I noticed we have test which is checking modal creation under container with the children count...
So if I dont have wrapper, 1 modal creates 3 elements under the container, which might be not that nice, cause if you have 2 modals, then is like 6 children elements and hard to get what is what..

Reply:
Shouldn't we fix the test then?